### PR TITLE
Review stable

### PR DESCRIPTION
**Pull request by milobot**
## Work in Progress
- Stochastic Gradient Descent optimizer
- Training SGD optimized model with data (using deeplearn.js)
- Uploading custom datasets using CSVs
## Issues
- Plots do not currently show legend key when only one plot is present
- Helpurls and tooltips are not present for most blocks
- Non-standardized block colors and organization - Miscellaneous category needs to be reworked

## Features implemented:
- Basic JS blocks + code generation
- UI for workspace, code execution, XML
- ML Features
     - Loading in-built datasets (Iris, California Housing dataset)
     - Plotting - Histogram, Scatter plots
     - Images - from URL, Webcam, Squeezenet Classifier
     - Math - Deeplearn.js Array1D, Scalar Arithmetic, Array from Text

### CODE REVIEW

**Comment by arjun-rao** [comment]
Stale pull request.
